pkg/http/randomizer.go

```diff
@@ -0,0 +1,18 @@
+package http
+
+import (
+	"net/http"
+	"sync/atomic"
+)
+
+// closeEvery is the chance that a request will be closed.
+const closeEvery = 10000 // every 1000 requests will be closed
+
+var counter uint64
+
+func MaybeCloseConnection(w http.ResponseWriter, r *http.Request) {
+	if atomic.AddUint64(&counter, 1)%closeEvery == 0 {
```

Comment:
I'm not sure this will enforce fairness very well vs per-connection counters, but I think this is fine to move forward.

I'm also a bit worried that under high request rates that we may "skip" closes, but I may misunderstand how AddUint64 works under the hood. My model is:

counter=9999
thread1: add 1
thread2: add 1
thread1: return 10001
thread1: return 10001

Reply:
For per connection ones if we go that way in the future, I've used the `ConnContext` field in `Server` to populate a uint initialized to zero which I have incremented and checked to see if it's time to kick that connection off.

Reply:
Atomic won't allow that.  After thread1 adds, it'll be 10000.  After thread2 add, it's 10002.